clarity/tests/nft-marketplace.test.ts

```diff
@@ -9,373 +9,369 @@ const deployer = accounts.get("deployer")!;
-        deployer
+        [Cl.contractPrincipal(deployer, "funny-dog"), Cl.bool(true)],
+        deployer,
       );
 
       // Mint an NFT for the user
-      const mintResponse = await simnet.callPublicFn(
+      const mintResponse = simnet.callPublicFn(
         "funny-dog",
         "mint",
         [Cl.principal(user1)],
-        deployer
+        deployer,
       );
       expect(mintResponse.result).toBeOk(Cl.uint(1));
 
       // Create listing
-      const listingResponse = await createListing(user1);
+      const listingResponse = createListing(user1);
       expect(listingResponse.result).toBeOk(Cl.uint(0)); // First listing should have ID 0
 
       // Verify listing details
-      const getListing = await simnet.callReadOnlyFn(
+      const { result } = simnet.callReadOnlyFn(
         "nft-marketplace",
         "get-listing",
         [Cl.uint(0)],
-        user1
+        user1,
       );
 
       // Check that the listing exists and has correct details
-      const listing = getListing.result.value as any;
-      expect(listing.data).toBeDefined();
-      expect(listing.data.maker).toEqual(Cl.principal(user1));
-      expect(listing.data['nft-asset-contract']).toEqual(Cl.contractPrincipal(deployer, 'funny-dog'));
-      expect(listing.data['token-id']).toEqual(Cl.uint(1));
-      expect(listing.data.expiry.type).toEqual(1); // uint type
-      expect(Number(listing.data.expiry.value)).toBeGreaterThan(Number(simnet.blockHeight)); // expiry should be in the future
-      expect(listing.data.price).toEqual(Cl.uint(100));
-      expect(listing.data.taker.type).toEqual(9); // none type
-      expect(listing.data['payment-asset-contract'].type).toEqual(9); // none type
+      expect(result).toBeSome(
+        Cl.tuple({
+          expiry: Cl.uint(35),
+          maker: Cl.principal(user1),
+          "nft-asset-contract": Cl.contractPrincipal(deployer, "funny-dog"),
+          "token-id": Cl.uint(1),
+          price: Cl.uint(100),
+          taker: Cl.none(),
+          "payment-asset-contract": Cl.none(),
+        }),
+      );
+
+      // Alternatively, the tuple value can be accessed directly
+      const listing = (result as SomeCV<TupleCV>).value;
+      expect(listing.value).toBeDefined();
+      expect(listing.value.maker).toEqual(Cl.principal(user1));
+      expect(listing.value["nft-asset-contract"]).toEqual(
+        Cl.contractPrincipal(deployer, "funny-dog"),
+      );
+      expect(listing.value["token-id"]).toEqual(Cl.uint(1));
+      expect(listing.value.expiry).toHaveClarityType(ClarityType.UInt);
+      const expiry = listing.value.expiry as SomeCV;
+      expect(Number(expiry.value)).toBeGreaterThan(Number(simnet.blockHeight));
+      expect(listing.value.price).toEqual(Cl.uint(100));
+      expect(listing.value.taker).toHaveClarityType(ClarityType.OptionalNone);
+      expect(listing.value["payment-asset-contract"]).toHaveClarityType(
+        ClarityType.OptionalNone,
+      );
```

Comment:
We can leave it for educational purpose, but realistically, the assertion above is more recommended